services/scanners/task_parser.go

```diff
@@ -213,16 +218,34 @@ func (p *ParserTask) parseBlockTransactions(block oasis.Block) (err error) {
 			receiver = raw.Body.Account.String()
 		}
 
+		reclaimAmount := quantity.NewQuantity()
+		//Get reclaim amount
+		if txType.Type() == dmodels.TransactionTypeReclaimEscrow {
+			stakeAccount, err := p.stakingAPI.Account(p.ctx, &stakingAPI.OwnerQuery{
+				Height: block.Header.Height,
+				Owner:  raw.Body.Account,
+			})
+			if err != nil {
+				return err
+			}
+
+			reclaimAmount, err = stakeAccount.Escrow.Debonding.StakeForShares(&raw.Body.Shares)
```

Comment:
That was my main question. We should use `stake.Account.Escrow.Active` of current block or block before?

Reply:
That is a good question.

@pro-wh, @kostko, if a reclaim escrow transaction happens at height `h`, would querying the corresponding staking account at height `h`, e.g. with something like:
```
stakeAccount, err := stakingAPI.Account(p.ctx, &stakingAPI.OwnerQuery{
    Height: <h>,
    Owner:  <owner>,
})
```
return `stakeAccount.Escrow.Active`'s `Balance` and `TotalShares` already reduced by the reclaimed amount/shares?

In other words, does querying account balances (general account, escrow accounts) for height `h` return the state before txns for height `h` are processed or the state after the txns are processed?

Reply:
> So, it would seem the right thing to do is to compute the reclaimed amount using stake.Account.Escrow.Active share pool?

This is would be the way to do it (although also depending on Tadej's later question about height. In the case that querying at height `h` gives the state after the transaction, then it would be correct to use `...Debonding.StakeForShares(<the number of debonding shares>)`, which is a different number from the number of "active escrow" shares from the transaction.

As for whether querying at height `h` includes transactions that "happen at height `h`," I don't have the answer to that. Legend has it that Tendermint was built for the rule that a query at height `h` refers to the state before the block of height `h` is processed. But I'm not sure all Oasis code follows that rule, or that it follows any rule consistently. I'm not even sure if the phrase "happens at height `h`" refers to the transaction being included in the block of height `h`.

Reply:
Ok, I manually checked what happened with account [oasis1qqs6ylpfurhf6gc9mw232fkmrt3d0673lyzc5xf2 ](https://oasismonitor.com/account/oasis1qqs6ylpfurhf6gc9mw232fkmrt3d0673lyzc5xf2) and transaction [2b1bce3218a30448d985003f8e1e6f367bf85c9e5d6c5397a0857bf3803886e8](https://oasismonitor.com/operation/2b1bce3218a30448d985003f8e1e6f367bf85c9e5d6c5397a0857bf3803886e8).

```
# Height: 3053632
$ oasis-node-dev stake account info --stake.account.address oasis1qqs6ylpfurhf6gc9mw232fkmrt3d0673lyzc5xf2
Balance:
  Total: 2024872399.487212065 ROSE
  Available: 3244190.999715 ROSE
  Active Delegations from this Account:
    Total: 2021628208.487497065 ROSE
    Delegations:

      ... trimmed ...

Nonce: 10

# Tx 2b1bce3218a30448d985003f8e1e6f367bf85c9e5d6c5397a0857bf3803886e8 at height 3053633:
# staking.Transfer
# From: oasis1qqs6ylpfurhf6gc9mw232fkmrt3d0673lyzc5xf2
# To: oasis1qzf23c8yv574qxwuahg439gj5yl8mhsuzsmna90p
# Amount: 2165000 ROSE

# Height: 3053633
$ oasis-node-dev stake account info --stake.account.address oasis1qqs6ylpfurhf6gc9mw232fkmrt3d0673lyzc5xf2
Balance:
  Total: 2022707399.487182065 ROSE
  Available: 1079190.999685 ROSE
  Active Delegations from this Account:
    Total: 2021628208.487497065 ROSE
    Delegations:

      ... trimmed ...

Nonce: 11
```

---

This example shows that querying the staking account at height `h` returns the state after the transactions at height `h` are processed.

Hence, as @pro-wh wrote, the correct thing would be to use to use `...Debonding.StakeForShares(<the number of debonding shares>)`. But note that **number of "debonding" shares** is a **different** number **from** the **number of "active escrow" shares** from the transaction.

So, the current implementation:
https://github.com/everstake/oasis-explorer/blob/ba6e321a9e553dcd9dfe2561c253581d3d2dc88d/services/scanners/task_parser.go#L232

would return wrong results.